src/Nethermind/Nethermind.JsonRpc.Test/Modules/DebugRpcModuleTests.TraceCallMany.cs

```diff
@@ -0,0 +1,486 @@
+                    To = to,
+                    Value = 1.Ether(),
+                    Gas = 21000
+                }
+            }
+        };
+
+        var result = ctx.DebugRpcModule.debug_traceCallMany(new[] { bundle }, BlockParameter.Latest);
+
+        result.Should().BeOfType<ResultWrapper<IReadOnlyList<GethLikeTxTrace[]>>>();
+        result.Data.Should().HaveCount(1);
+        result.Data[0].Should().HaveCount(1);
+        result.Data[0][0].Should().NotBeNull();
+    }
+
+    [Test]
+    public async Task Debug_traceCallMany_with_multiple_bundles()
+    {
+        using Context ctx = await Context.Create();
+
+        Address from = Build.An.Address.TestObject;
+        Address to1 = TestItem.AddressC;
+        Address to2 = TestItem.AddressD;
+        UInt256 balance = 100.Ether();
+
+        await ctx.Blockchain.AddFunds(from, balance);
+
+        var bundles = new[]
+        {
+            new TransactionBundle
+            {
+                Transactions = new[]
+                {
+                    new LegacyTransactionForRpc
+                    {
+                        From = from,
+                        To = to1,
+                        Value = 1.Ether(),
+                        Gas = 21000
+                    }
+                }
+            },
+            new TransactionBundle
+            {
+                Transactions = new[]
+                {
+                    new LegacyTransactionForRpc
+                    {
+                        From = from,
+                        To = to2,
+                        Value = 2.Ether(),
+                        Gas = 21000
+                    }
+                }
+            }
+        };
+
+        var result = ctx.DebugRpcModule.debug_traceCallMany(bundles, BlockParameter.Latest);
+
+        result.Should().BeOfType<ResultWrapper<IReadOnlyList<GethLikeTxTrace[]>>>();
+        result.Data.Should().HaveCount(2);
+        result.Data[0].Should().HaveCount(1);
+        result.Data[1].Should().HaveCount(1);
+    }
+
+    [Test]
+    public async Task Debug_traceCallMany_with_multiple_transactions_per_bundle()
+    {
+        using Context ctx = await Context.Create();
+
+        Address from = Build.An.Address.TestObject;
+        Address to1 = TestItem.AddressC;
+        Address to2 = TestItem.AddressD;
+        UInt256 balance = 100.Ether();
+
+        await ctx.Blockchain.AddFunds(from, balance);
+
+        var bundle = new TransactionBundle
+        {
+            Transactions = new[]
+            {
+                new LegacyTransactionForRpc
+                {
+                    From = from,
+                    To = to1,
+                    Value = 1.Ether(),
+                    Gas = 21000
+                },
+                new LegacyTransactionForRpc
+                {
+                    From = from,
+                    To = to2,
+                    Value = 2.Ether(),
+                    Gas = 21000
+                }
+            }
+        };
```

Comment:
If you are reusing same or similar bundles in different tests, maybe make a method that generates them?